src/Workspaces/Core/Portable/Workspace/Solution/SolutionState_Checksum.cs

```diff
@@ -100,7 +109,7 @@ void AddReferencedProjects(HashSet<ProjectId> result, ProjectId projectId)
                     // host to remove the reference.  We do not expose this through the full Solution/Project which
                     // filters out this case already (in Project.ProjectReferences). However, becausde we're at the
                     // ProjectState level it cannot do that filtering unless examined through us (the SolutionState).
```

Comment:
```suggestion
                    // ProjectState level it cannot do that filtering unless examined through us (the SolutionCompilationState).
```

Although I'm not really sure the difference matters here.